src/psyclone/tests/conftest.py

```diff
@@ -143,18 +142,36 @@ def infra_compile(tmpdir_factory, request):
     GOcean1p0Build(tmpdir)
 
 
-@pytest.fixture(scope="session")
-def parser():
+@pytest.fixture(name="_session_parser", scope="session")
+def _session_parser():
     '''
     Creates and returns an fparser object. Since this is expensive we only
-    do this once per test session (scope="session" above).
+    do this once per test session (scope="session" above). This fixture is
+    only intended to be used in the 'public' fixture `parser` below.
+
+    # TODO #1188 - move this to tests/psyir/frontend/conftest.py
+
+    '''
+    return ParserFactory().create(std="f2008")
+
+
+@pytest.fixture(scope="function")
+def parser(_session_parser):
+    '''
+    Returns the session fparser object but clears any existing symbol tables
+    before doing so.
+
+    TODO #1188 as part of isolating fparser usage to the PSyIR frontend, this
```

Comment:
I would suggest a colon ":" after "1188", i.e. `TODO #1188: as part of ...`. See also the comment about prefixing TODO with "#" in line 152.

Reply:
I've gone for '-' to be consistent with the previous TODO.